scipy/_lib/_uarray/_uarray_dispatch.cxx

```diff
@@ -15,13 +15,43 @@
 
 namespace {
 
+template <typename T>
+class immortal {
+    alignas(T) std::byte storage[sizeof(T)];
+
+public:
+    template <typename... Args>
+    immortal(Args&&... args) {
+       // Construct new T in storage
+       new(&storage) T(std::forward<Args>(args)...);
+    }
+    ~immortal() {
+        // Intentionally don't call destructor
+    }
+
+    // T* get() const { return reinterpret_cast<T*>(&storage); }
+    const T* get() const { return reinterpret_cast<const T*>(&storage); }
+    const T* get_const() const { return reinterpret_cast<const T*>(&storage); }
+
+
+    const T* operator ->() const { return get(); }
+    // T* operator ->() const { return get(); }
+
+    friend bool operator==(const immortal<T> & lhs, const immortal<T> & rhs) {
+      return lhs.get() == rhs.get();
+    }
```

Comment:
Sorry, I missed a few things in my earlier comment. I think this should be more complete
```suggestion
    T* get() { return reinterpret_cast<T*>(&storage); }
    const T* get() const { return reinterpret_cast<const T*>(&storage); }
    const T* get_const() const { return reinterpret_cast<const T*>(&storage); }

    const T* operator ->() const { return get(); }
    T* operator ->() { return get(); }

    T& operator*() { return *get(); }
    const T& operator*() const { return *get(); }
```